lsp-pyright.el

```diff
@@ -228,6 +256,10 @@ Current LSP WORKSPACE should be passed in."
    (,(concat lsp-pyright-langserver-command ".disableOrganizeImports") lsp-pyright-disable-organize-imports t)
    (,(concat lsp-pyright-langserver-command ".disableTaggedHints") lsp-pyright-disable-tagged-hints t)
    (,(concat lsp-pyright-langserver-command ".typeCheckingMode") lsp-pyright-type-checking-mode)
+   ("basedpyright.analysis.inlayHints.variableTypes" lsp-pyright-basedpyright-inlay-hints-variable-types t)
```

Comment:
Does Pyright ignore settings it doesn't recognize? Have you tested this with Pyright? If this works, does Pyright make any guarantees it'll always ignore settings it doesn't recognize?

Reply:
@wyuenho 

As you can see the S expression, `lsp-register-custom-settings` does NOT associate settings with `lsp-client`.

In order to get configuration, each client executable sends `workspace/configuration` request. `lsp-mode` send the configuration ONLY if `pyright` send request to get `basedpyright.*`

In the other word, if `pyright` executable should require even `java` section, `lsp-mode` pass the setting for `lsp-java` to `pyright`.

If we need the proof that this configuration does not affect `pyright`, we should see all `lsp-client` in `lsp-mode`, since all `lsp-register-custom-settings` expressions can affect `pyright`.

Reply:
In conclusion, as long as `pyright` do not send request to get `basedpyright.*`, `lsp-mode` never send this configuration.

Reply:
Thanks for the clarification, just to aid my understanding, it's actually the server that sends the `workspace/configuration` request, not the client right? This seems to be what the latest version of the protocol requires.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_configuration

Reply:
Sorry, I was confused. You are right. The server sends `workspace/configuraiton` request.